### ISSUE TITLE
Labeling of the `shift` key

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2841]
In the codebase we refer to the Shift key as `"shift"`, whereas in the documentation we refer to it as `"s"`.

Find an example below.

Help interface:

![2023-03-10_16:50:09](https://user-images.githubusercontent.com/45483512/224347941-be1af614-0f38-4d78-8946-a187e5e1b5e8.png)

Codebase:

```lisp
"shift-left" 'backward-char-with-selection
"shift-right" 'forward-char-with-selection
"shift-up" 'backward-line-with-selection
"shift-down" 'forward-line-with-selection
```

This means that the codebase is actually more approachable than the Help interface. 

I suggest referring to the Shift key as `"shift"` everywhere.

@lansingthomas, does it make sense?

**Comment by Ambrevar** [comment]
Yes!

**Comment by aartaka** [comment]
I'd rather refer to it as `"s"` for consistency. `"Ctrl"`, `"shift"`, and other fruits of decadent CUA-ish civilization can wait until https://github.com/atlas-engineer/nkeymaps/issues/4.

**Comment by lansingthomas** [comment]
### great idea!
I am in favor of `shift` for myself
I am in favor of `shift` for most of our users (though I have not asked them about it)
I am in favor of `s` for our beloved emacs wizards IF they already know it and prefer it.

This could be very impactful for people learning our system!

### Before anything is built, let's answer these questions.
- [ ] Would it make sense to have `shift` displayed for the default CUA binding mode and then `s` display when `emacs mode` is enabled? (this way anyone who has successfully changed their launch configuration to emacs mode will see the familiar `s`, and the rest will see a `shift`.
- [ ] Could we have [noob keybindings display mode](https://github.com/atlas-engineer/nkeymaps/issues/4) as a configurable setting in the `common settings` section?

### reasoning: 
- declaring `shift` helps new users understand when to use lowercase `s` or uppercase `S` (shift + lowercase `s`) and or `shift`
- I understand that many of our emacs users will be familiar with `s`, but I am not convinced that they prefer it to the `shift` -- AND i know they will recognize `shift` 
    - [ ] do they prefer the little `s`?

**Comment by jmercouris** [comment]
I don't think anyone prefers the little "s". I know many Emacs users get tripped up on "shift" vs "super" all the time, which in Emacs parlance is "S" and "s".

**Comment by aartaka** [comment]
I prefer the little `s`, because it's concise and consistent with the rest of our notation. 

I'd also say that Emacs and VI modes SHOULD (yes, I've read the RFC on requirement levels) display shift as `s` (for the reasons above), while CUA mode MUST display shift as `shift`, to make it recognizable enough.

**Comment by aadcg** [comment]
> I don't think anyone prefers the little "s". I know many Emacs users get tripped up on "shift" vs "super" all the time, which in Emacs parlance is "S" and "s".

I second this view. 

On the topic of supporting multiple notations, I think it's not wise. There are two dimensions here: 

1. How keybindings are specified programmatically (codebase, config files);
2. How keybindings are formatted when shown in the Help system (aesthetics).

With regards to 1., I don't see any benefit in supporting multiple representations for the same key. For example, what would  be the benefit of having `"C-shift-tab"` and `"C-s-tab"` refer to the same keybinding? Sometimes that needs to be done, but that's for backwards compatibility reasons (not relevant here). This explains why I oppose supporting `"s"` as an additional representation to specify the Shift key in keybindings programmatically. In the current setting, all users specify their keybindings featuring the Shift key uniformly (via `"shift"`) and that's intelligible for users of the  CUA, Emacs or VI keyschemes.

With regards to 2., I see no good reason to move away from the programmatical representation. Notice that CUA users also write their own config files! I.e., they also need to learn the programatical notation for our keybindings. If we break this symmetry (aesthetical/programatical representation) we're actually raising the burden on the user. Say that we  were to display `"C-a"` as `"Control-a"` or `"CTRL-a"`. Then the user has to keep this duality in this mind that there's the aesthetical representation of the keybinding and how it is specified programatically. 

In conclusion, I still think that my initial proposal is superior. Notice that making the change I propose is trivial.

**Comment by aartaka** [comment]
> With regards to 2., I see no good reason to move away from the programmatical representation. Notice that CUA users also write their own config files! I.e., they also need to learn the programatical notation for our keybindings.

Then we can have just the verbose notation for CUA people, like `control-m`, and that would still work as a config value. We can even add case-insensitivity to `nkeymaps`, to allow a notation like `Control-m`. And we can even add parsing of `+` as a separator. And!... And!... Okay, I'll stop at that case-insensitivity and plus, those should be more than enough for conventionality.

**Comment by aadcg** [comment]
There's a mistake in my comment above. Programatically, we support both `"shift"` and `"s"`. I.e. it's equivalent to specify `"shift-space"` and `"s-space"`. I was not aware of this! I thought the only valid way was to specify it as `"shift"`.

Given this new piece of information, I suggest that both programatically and aesthetically we should stick to one notation. But I still maintain what I've mentioned above:

> Say that we were to display "C-a" as "Control-a" or "CTRL-a". Then the user has to keep this duality in this mind that there's the aesthetical representation of the keybinding and how it is specified programatically.

**Comment by aadcg** [comment]
> Then we can have just the verbose notation for CUA people, like control-m, and that would still work as a config value.

At this point, I think we should be very good at doing each thing in a unified way. The more degrees of freedom you add, the more you have to maintain, and the more you can error. 

Regardless, I agree that all aesthetical representation (while I advocate that we should have a single one) should be a valid way of specifying keybindings in config files. Again, I was unaware that `"C-a"` and `"control-a"` are both valid in config files.

**Comment by aartaka** [comment]
> The more degrees of freedom you add, the more you have to maintain, and the more you can error.

I disagree, but that (Chaos/System Science, emergent behavior, evolving programs) is a topic for another time.
